runtime/memprof.c

```diff
@@ -629,6 +629,9 @@ static memprof_orphan_table_t orphans = NULL;
 /* lock controlling access to `orphans` variable */
 static caml_plat_mutex orphans_lock = CAML_PLAT_MUTEX_INITIALIZER;
 
+/* Flag indicating non-NULL orphans. Only modified when holding orphans_lock. */
+static atomic_uintnat orphans_flag;
```

Comment:
Use a name to describe the condition when this flag is true (e.g. `orphans_present`).

Reply:
Good idea, done.